### ISSUE TITLE
Error: StackOverflowError

### ISSUE THREAD

**Issue by sugguy** [d4rken-org/permission-pilot#173]
- Miui 13.0.6
- Android 12
- App : v1.1.5-rc0 (10105000) ~ b93dce8/FOSS/ RELEASE 

When launching the app, I get an error window :

```
Error: StackOverflowError 

stack size 1039KB 

```

**Comment by d4rken** [comment]
Are you able to grab a stacktrace?
Is more displayed than this?

**Comment by sugguy** [comment]
> this

I don't know how to do that.
And yes only that message shows on Apps an Overview tabs.
And something I just noticed: there is a loading

**Comment by d4rken** [comment]
In the app settings there is an option to record a debug log, can you use that to record the error and upload the log?

**Comment by sugguy** [comment]
[eu.darken.myperm_logfile_1677083058017.log.zip](https://github.com/d4rken-org/permission-pilot/files/10806675/eu.darken.myperm_logfile_1677083058017.log.zip)

**Comment by d4rken** [comment]
Hm i tried to reproduce this but can't. Recursion doesn't happen for me.
Feels like a bug in the on device app optimization, which would be pretty :exploding_head:.

I'll remove the default interface implementation then it should work.

```
	at eu.darken.myperm.apps.core.container.SecondaryProfilePkg.isSystemApp(SecondaryProfilePkg.kt:72)
	at eu.darken.myperm.apps.core.container.SecondaryProfilePkg.isSystemApp(SecondaryProfilePkg.kt:72)
	at eu.darken.myperm.apps.core.container.SecondaryProfilePkg.isSystemApp(SecondaryProfilePkg.kt:72)
	at eu.darken.myperm.main.ui.overview.OverviewFragmentVM$summaryData$1.invokeSuspend(OverviewFragmentVM.kt:69)
	at eu.darken.myperm.main.ui.overview.OverviewFragmentVM$summaryData$1.invoke(Unknown Source:11)
	at eu.darken.myperm.main.ui.overview.OverviewFragmentVM$summaryData$1.invoke(Unknown Source:6)
```

**Comment by d4rken** [comment]
Fix will be in v1.1.6 (currently building).

Thanks for the report!

**Comment by sugguy** [comment]
I confirm that the issue is gone with [v1.1.6-rc0](https://github.com/d4rken-org/permission-pilot/releases/tag/v1.1.6-rc0)
